src/pages-and-resources/live/Settings.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as Yup from 'yup';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { CardGrid } from '@edx/paragon';
+import AppCard from './AppCard';
+import FormikControl from '../../generic/FormikControl';
+import { useAppSetting } from '../../utils';
+import AppExternalLinks from '../discussions/app-config-form/apps/shared/AppExternalLinks';
+import AppSettingsModal from '../app-settings-modal/AppSettingsModal';
+import messages from './messages';
+
+function LiveSettings({
+  intl,
+  onClose,
+}) {
+  const [liveConfiguration, saveSettings] = useAppSetting('liveConfiguration');
+
+  const handleSettingsSave = async (values) => {
+    saveSettings(values);
+  };
+
+  const selectProvider = () => {};
+
+  return (
+    <AppSettingsModal
+      appId="live"
+      title={intl.formatMessage(messages.heading)}
+      enableAppHelp={intl.formatMessage(messages.enableLiveHelp)}
+      enableAppLabel={intl.formatMessage(messages.enableLiveLabel)}
+      learnMoreText={intl.formatMessage(messages.enableLiveLink)}
+      onClose={onClose}
+      initialValues={{
+        consumerKey: liveConfiguration?.consumerKey || '',
+        consumerSecret: liveConfiguration?.consumerSecret || '',
+        launchUrl: liveConfiguration?.launchUrl || '',
+        launchEmail: liveConfiguration?.launchEmail || '',
+        piiShareUsername: liveConfiguration?.piiSharing ? liveConfiguration.piiShareUsername : undefined,
+        piiShareEmail: liveConfiguration?.piiSharing ? liveConfiguration.piiShareEmail : undefined,
+      }}
+      validationSchema={{
+        enabled: Yup.boolean(),
+        consumerKey: Yup.string().required(intl.formatMessage(messages.consumerKeyRequired)),
+        consumerSecret: Yup.string().required(intl.formatMessage(messages.consumerSecretRequired)),
+        launchUrl: Yup.string().required(intl.formatMessage(messages.launchUrlRequired)),
+        launchEmail: Yup.string().required(intl.formatMessage(messages.launchEmailRequired)),
+      }}
```

Comment:
create validation schema above and pass here for clean code.

Reply:
Sure.